ci/scan_flattened_deps.py

```diff
@@ -0,0 +1,340 @@
+#!/usr/bin/env python3
+#
+# Usage: scan_flattened_deps.py --flat-deps <flat DEPS file> --output <vulnerability report>
+#
+# This script parses the flattened, fully qualified dependencies,
+# and uses the OSV API to check for known vulnerabilities
+# for the given hash of the dependency
+
+import argparse
+import json
+import os
+import sys
+import subprocess
+import time
+import requests
+
+SCRIPT_DIR = os.path.dirname(sys.argv[0])
+CHECKOUT_ROOT = os.path.realpath(os.path.join(SCRIPT_DIR, '..'))
+UPSTREAM_PREFIX = 'upstream_'
+
+HELP_STR = 'To find complete information on this vulnerability, navigate to '
+# TODO -- use prefix matching for this rather than always to OSV
+OSV_VULN_DB_URL = 'https://osv.dev/vulnerability/'
+DEPS = os.path.join(CHECKOUT_ROOT, 'DEPS')
+
+failed_deps = []  # deps which fail to be be cloned or git-merge based
```

Comment:
This is not necessarily requiring action. Further down the line if we want a way to automate dependency rolling, it will be helpful to understand which deps cannot be scanned with this process.